File: src/main/java/net/theevilreaper/vulpes/api/repository/ItemRepository.java
package net.theevilreaper.vulpes.api.repository;

import io.micronaut.data.mongodb.annotation.MongoRepository;
import io.micronaut.data.repository.PageableRepository;
import net.theevilreaper.vulpes.api.model.ItemModel;

@MongoRepository
public interface ItemRepository extends PageableRepository<ItemModel, String> {
}

File: src/main/kotlin/net/theevilreaper/vulpes/api/repository/ItemRepository.kt
package net.theevilreaper.vulpes.api.repository

import net.theevilreaper.vulpes.api.model.ItemModel
import org.springframework.data.mongodb.repository.MongoRepository

/**
 * @author theEvilReaper
 * @version 1.0.0
 * @since
 **/

interface ItemRepository : MongoRepository<ItemModel, String>